components/library/src/content/file_info.rs

```diff
@@ -26,15 +34,48 @@ pub fn find_content_components<P: AsRef<Path>>(path: P) -> Vec<String> {
     components
 }
 
-/// Struct that contains all the information about the actual file
+/// Return a `(name, maybe_lang)` tuple if name has a possible language in it
+pub fn get_language<S: AsRef<str>>(name: S) -> (String, Option<String>) {
+    // Go with the assumption that no one is using '.' in filenames (regardless of whether the site
+    // is multilingual)
+    if !name.as_ref().contains('.') {
+        return (name.as_ref().to_string(), None);
+    }
+
+    let parts: Vec<String> = name.as_ref().splitn(2, '.').map(ToString::to_string).collect();
+    (parts[0].clone(), Some(parts[1].clone()))
+}
+
+/// Return a `(name, date)` tuple if name has a date in it
+///
+/// This only makes sense for pages, not sections.
+pub fn get_date<S: AsRef<str>>(name: S) -> (String, Option<String>) {
+    if let Some(ref caps) = RFC3339_DATE.captures(name.as_ref()) {
+        return (
+            caps.name("name").unwrap().as_str().to_string(),
+            Some(caps.name("datetime").unwrap().as_str().to_string()),
+        );
+    }
+    (name.as_ref().to_string(), None)
+}
+
+/// Strip `{date}{_,-}` prefix and `.{lang}` suffix from name
+pub fn clear_filename<S: AsRef<str>>(name: S) -> String {
```

Comment:
I don't like the name but I can't think of anything better gith now.

Reply:
`strip` as in my comment?